src/Component/BlogPage.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react'
+import './BlogPage.css'
+
+import BlogMiddleComp from './BlogMiddleComp'
+import { data } from '../data'
+function BlogPage() {
+
+    const [searchTerm, setSearchTerm] = useState('');
+    const handleSearch = (e) => {
+        setSearchTerm(e.target.value)
+        console.log(searchTerm)
+    }
+
+    const searchFilter = () => {
+
+        let arr = data.filter(function (d) {
+            // console.log(d.name)
+            if (d.name != undefined) {
```

Comment:
Start using eslint in your code. This should be `!==`. Also, you shouldn't be comparing to `undefined`. Search for how to check for string empty

Reply:
What if `d.name` was an object instead of string? You would get an exception below